### DISCUSSION TITLE
Keys in comments

### DISCUSSION THREAD

**Discussion by greg-vstorm** [WeblateOrg/weblate#6434] [Q&A] [answered]
Hi, I have a question about comments before translations. I noticed that comments can be used as keys
Source code is PHP (Laravel) and looks like this:

```php
return [
           // advices
            'advices'      => [
                // towards
                'tw' => [
                    // generic
                    'generic' => [
                        // male
                        'male'    => 'Adviezen die [hij] geeft',
                        // female
                        'female'  => 'Adviezen die [zij] geeft',
                        // neutral
                        'neutral' => 'Adviezen die [hij of zij] geeft',
                    ],
                    // bullit1
                    'bullit1' => [
                        // male
                        'male'    => 'Als [hij] iemand advies geeft, zal [hij] {misschien} zoiets aanraden als:',
                        // female
                        'female'  => 'Als [zij] iemand advies geeft, zal [zij] {misschien} zoiets aanraden als:',
                        // neutral
                        'neutral' => 'Als [hij of zij] iemand advies geeft, zal [hij of zij] {misschien} zoiets aanraden als:',
                    ],
                ],
            ],
];
```

When translating, in the right sidebar is "Source string comment" but includes just one or two last comments in the hierarchy. Same "key" above translated text - on dark background. For example "//male" or "//bullit1 // male"
How can I show full path like "// advices // towards // generic // male" ?
Source code is generated from imported .xlsx files, if this is not possible I can make long comments (with full path) above each string.

**Comment by nijel** [comment, marked as answer]
Comment concatenation in the structure is not implemented in the translate-toolkit, but patches are welcome there.

Why don't you translate the xlsx file directly?

**Reply by greg-vstorm** [reply]
Xlsx file is used only as a source for nl>en translation. For translations to other languages, English will be used as a source.
